runtime/common/src/crowdloan/migration.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_support::generate_storage_alias;
+
+/// Migrations for using fund index to create fund accounts instead of para ID.
+pub mod crowdloan_index_migration {
+	use super::*;
+
+	// The old way we generated fund accounts.
+	fn old_fund_account_id<T: Config>(index: ParaId) -> T::AccountId {
+		T::PalletId::get().into_sub_account(index)
+	}
+
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+		// `NextTrieIndex` should have a value.
+		generate_storage_alias!(Crowdloan, NextTrieIndex => Value<FundIndex>);
+		let next_index = NextTrieIndex::get().unwrap_or_default();
+		ensure!(next_index > 0, "Next index is zero, which implies no migration is needed.");
+
+		log::info!(
+			target: "runtime",
+			"next trie index: {:?}",
+			next_index,
+		);
+
+		// Each fund should have some non-zero balance.
+		for (para_id, fund) in Funds::<T>::iter() {
+			let old_fund_account = old_fund_account_id::<T>(para_id);
+			let total_balance = CurrencyOf::<T>::total_balance(&old_fund_account);
+
+			log::info!(
+				target: "runtime",
+				"para_id={:?}, old_fund_account={:?}, total_balance={:?}, fund.raised={:?}",
+				para_id, old_fund_account, total_balance, fund.raised
```

Comment:
```suggestion
				"para_id={:?}, old_fund_account={:?}, total_balance={:?}, fund.raised={:?}, total_balance.cmp(fund.raised)={:?}",
				para_id, old_fund_account, total_balance, fund.raised, fund.raised.cmp(&total_balance),
```

This is what I was using locally to get 

```
2022-02-07 12:35:44.154  INFO main runtime: para_id=Id(2000), old_fund_account=6d6f646c70792f6366756e64d007000000000000000000000000000000000000 (13UVJyLn...), total_balance=325159802333576263, fund.raised=325159802323576263, total_balance.cmp(fund.raised)=Less